packages/next-intl/src/middleware/middleware.tsx

```diff
@@ -32,6 +33,9 @@ export default function createMiddleware(config: MiddlewareConfig) {
   const matcher: Array<string> | undefined = (config as any)._matcher;
 
   return function middleware(request: NextRequest) {
+    // if Next.js behind a reverse proxy, it cannot get the correct host. ex) localhost, 172.x.x.x
+    // We need to set the host here, because Next.js doesn't do it for us
+    request.nextUrl.host = getHost(request.headers) ?? request.nextUrl.host;
```

Comment:
If possible, I'd like to avoid overwriting the incoming request. If the user [composes the middleware](https://next-intl-docs.vercel.app/docs/routing/middleware#composing-other-middlewares), this could lead to surprises.

Reply:
This seems to be a new issue arising in Next.js version ^13.4.13. Since it might not be a problem that needs to be addressed within this library, I will remove it for now.

Reply:
In previous versions, the host of request.url was pointing to localhost, allowing for correct rewriting or redirection. However, it seems that the behavior has changed in version ^13.4.13.

Reply:
Interesting. Do you have a link to the corresponding change or a bug report on the Next.js side by chance?

Reply:
It seems like this might be related to https://github.com/vercel/next.js/issues/54133.